lib/mix/tasks/compile_asn1.ex

```diff
@@ -0,0 +1,38 @@
+defmodule Mix.Tasks.Compile.Asn1 do
+  @moduledoc """
+  This task compiles all ASN.1 files (in `asn1/*.asn`) into erlang sources into `src` directory.
+  After that Erlang compiler (task Compile.Erlang) shall be run to compile them into .beam binaries.
+  """
+  @shortdoc "Compiles ASN.1 files to Erlang sources"
+
+  use Mix.Task
+
+  @asn1_src "asn.1"
+  @erl_src "src"
+
+  @spec run(term) :: :ok
+  def run(_) do
+    :ok = File.mkdir_p!(@erl_src)
+    for file <- ls_r!(@asn1_src) do
+      case Path.extname(file) do
```

Comment:
I would argue with that, but luckily I don't have to, since I've just removed this case along with `ls_r!/1` anyway.